Python/SVY21.py

```diff
@@ -143,17 +148,50 @@ def computeLatLon(self, N, E):
         # Compute Latitude
         latFactor = tPrime / (self.k * rhoPrime)
         latTerm1 = latFactor * ((Eprime * x) / 2)
-        latTerm2 = latFactor * ((Eprime * x3) / 24) * ((-4 * psiPrime2) + (9 * psiPrime) * (1 - tPrime2) + (12 * tPrime2))
-        latTerm3 = latFactor * ((Eprime * x5) / 720) * ((8 * psiPrime4) * (11 - 24 * tPrime2) - (12 * psiPrime3) * (21 - 71 * tPrime2) + (15 * psiPrime2) * (15 - 98 * tPrime2 + 15 * tPrime4) + (180 * psiPrime) * (5 * tPrime2 - 3 * tPrime4) + 360 * tPrime4)
+        latTerm2 = latFactor * ((Eprime * x3) / 24) * (
+                (-4 * psiPrime2) + (9 * psiPrime) * (1 - tPrime2) + (12 * tPrime2))
+        latTerm3 = latFactor * ((Eprime * x5) / 720) * (
+                (8 * psiPrime4) * (11 - 24 * tPrime2) - (12 * psiPrime3) * (21 - 71 * tPrime2) + (
+                15 * psiPrime2) * (15 - 98 * tPrime2 + 15 * tPrime4) + (180 * psiPrime) * (
+                        5 * tPrime2 - 3 * tPrime4) + 360 * tPrime4)
         latTerm4 = latFactor * ((Eprime * x7) / 40320) * (1385 - 3633 * tPrime2 + 4095 * tPrime4 + 1575 * tPrime6)
         lat = latPrime - latTerm1 + latTerm2 - latTerm3 + latTerm4
 
         # Compute Longitude
         secLatPrime = 1. / math.cos(lat)
         lonTerm1 = x * secLatPrime
         lonTerm2 = ((x3 * secLatPrime) / 6) * (psiPrime + 2 * tPrime2)
-        lonTerm3 = ((x5 * secLatPrime) / 120) * ((-4 * psiPrime3) * (1 - 6 * tPrime2) + psiPrime2 * (9 - 68 * tPrime2) + 72 * psiPrime * tPrime2 + 24 * tPrime4)
+        lonTerm3 = ((x5 * secLatPrime) / 120) * ((-4 * psiPrime3) * (1 - 6 * tPrime2) + psiPrime2 * (
+                9 - 68 * tPrime2) + 72 * psiPrime * tPrime2 + 24 * tPrime4)
         lonTerm4 = ((x7 * secLatPrime) / 5040) * (61 + 662 * tPrime2 + 1320 * tPrime4 + 720 * tPrime6)
         lon = (self.oLon * math.pi / 180) + lonTerm1 - lonTerm2 + lonTerm3 - lonTerm4
 
-        return (lat / (math.pi / 180), lon / (math.pi / 180))
+        #X IS LON AND Y IS LAT
+        return (lon / (math.pi / 180), lat / (math.pi / 180))
+
+
+
+cv = SVY21()
+with open(argv[2], "a") as output:
+    write = csv.writer(output)
+    write.writerow(["X", "Y"])
+output.close()
+
+with open(argv[1], 'r') as csvfile:
+    input = csv.reader(csvfile)
+    next(input)
+    for row in input:
+        # print(row[0], row[1])
+        x = float(row[0])
+        y = float(row[1])
+        result = cv.computeLatLon(y, x)
+
+        with open(argv[2], "a") as output:
+            write = csv.writer(output)
+            write.writerow(result)
+
+        output.close()
+
+    print("Done! Check output.csv file in folder")
+    
+csvfile.close()
```

Comment:
This file was meant to be imported as a module, so this chunk of code should at least be wrapped in a main block, otherwise it breaks on import. It may even be better to write this as a separate tools script.

Reply:
Thanks for the feedback, though I am confused as to how to implement this as a tools script? don't have much experience at it...

Reply:
Oh, sorry, what I meant was that you could possibly write this csv conversion code in a separate file (maybe named something like `csv-converter.py`), and have it call `import SVY21 from SVY21` at the top.
This will help with separation of concerns if you ever want to add more functionality to your tool.